### PR TITLE
FAPI: Improve integration tests.

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
* The asserts which are activated by --enable-debug are replaced with ASSERT, which
  is always active, to ensure that for FAPI integration tests the asserts are
  always fulfilled.

* For JSON encoded results, if possible, the complete JSON value will be checked.
  If parts of the JSON object are variable only selected sub fields will be
  checked.

* The result of Fapi_List will be compared with an expected list of objects for
  typical use cases.

Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

### CODE REVIEW

File: test/integration/main-fapi.c [hunk 1]
```diff
@@ -43,6 +45,124 @@ bool file_exists (char *path) {
   return (stat (path, &buffer) == 0);
 }
 
+/* Determine integer number from json object. */
+static int64_t
+get_number(json_object *jso)
+{
```
Comment (cplappert) [inline test/integration/main-fapi.c:51]: "{" should be one line above

File: test/integration/test-fapi.h [hunk 2]
```diff
@@ -5,13 +5,127 @@
...
+             ASSERT(jso2); \
+             if (cmp_jso(jso1, jso2)) { \
+                break; \
+             } \
+             json_object_put(jso2); \
+        } \
+        if (i >=  n) { \
+            json_object_put(jso1);       \
+            json_object_put(jso2);          \
+            LOG_ERROR("Mismatch" ); \
```
Comment (cplappert) [inline test/integration/test-fapi.h:122]: Unnecessary Whitespace

**Review by cplappert — CHANGES_REQUESTED** [review]
I found a few minor things. Mostly formatting issues.
Otherwise, LGTM.

File: test/integration/test-fapi.h (hunk 2 above)
Comment (JuergenReppSIT) [reply test/integration/test-fapi.h:117]: @cplappert Thank you for the review.  I just pushed the changes.

**Review by cplappert — APPROVED** [review]
